src/views/User.vue

```diff
@@ -0,0 +1,227 @@
+        />
+      </v-row>
+
+      <v-row class="mx-4 mt-0 mb-2">
+        <ToggleOptOut
+          term-id="all"
+          section-id="all"
+          :instructor-uid="user.uid"
+          :initial-value="!!user.hasOptedOutForAllTerms"
+          label="for all semesters"
+          :before-toggle="() => (refreshingCourses = true)"
+          :on-toggle="refreshUser"
+        />
+      </v-row>
+
+      <Spinner v-if="refreshingCourses" />
+
+      <div v-if="!refreshingCourses && eligibleCourses.length" id="user-courses-eligible">
+        <CoursesDataTable
+          class="pt-5"
+          :courses="eligibleCourses"
+          :include-room-column="true"
+          :include-opt-out-column-for-uid="user.uid"
+          :message-for-courses="summarize(eligibleCourses)"
+          :refreshing="false"
+        />
+      </div>
+
+      <div v-if="!refreshingCourses && ineligibleCourses.length" id="user-courses-ineligible">
+        <h2 class="px-4">Courses not in a course capture classroom</h2>
+        <CoursesDataTable
+          class="pt-5"
+          :courses="ineligibleCourses"
+          :include-room-column="true"
+          :message-for-courses="summarize(ineligibleCourses)"
+          :refreshing="false"
+        />
+      </div>
+    </v-card>
+
+    <v-card v-if="currentUser.isAdmin" variant="outlined" class="elevation-1 mt-4">
+      <v-card-title>Notes</v-card-title>
+
+      <v-card-text v-if="!isEditingNote" id="note-body">
+        {{ user.note || 'No notes.' }}
+      </v-card-text>
+
+      <v-card-actions v-if="!isEditingNote" class="px-4 pb-4">
+        <v-btn
+          id="btn-edit-note"
+          variant="elevated"
+          :disabled="isUpdatingNote"
+          @click="editNote"
+        >
+          Edit
+        </v-btn>
+        <v-btn
+          v-if="user.note"
+          id="btn-delete-note"
+          class="mx-3"
+          variant="elevated"
+          :disabled="isUpdatingNote"
+          @click="deleteNote"
+        >
+          Delete
+        </v-btn>
+      </v-card-actions>
+
+      <v-card-text v-if="isEditingNote">
+        <v-textarea
+          id="note-body-edit"
+          v-model="noteBody"
+          variant="outlined"
+          hide-details="auto"
+          density="compact"
+          placeholder="Enter note text"
+        />
+      </v-card-text>
+
+      <v-card-actions v-if="isEditingNote" class="px-4 pb-4">
+        <v-btn
+          id="btn-save-note"
+          color="success"
+          :disabled="!noteBody || isUpdatingNote"
+          variant="elevated"
+          @click="saveNote"
+        >
+          Save
+        </v-btn>
+        <v-btn
+          id="btn-cancel-note"
+          class="mx-3"
+          variant="elevated"
+          :disabled="isUpdatingNote"
+          @click="cancelNote"
+        >
+          Cancel
+        </v-btn>
+      </v-card-actions>
+    </v-card>
+  </div>
+
+  <!-- fallback spinner while loading -->
+  <Spinner v-else />
```

Comment:
The Spinner component is included in BaseView, so you don't need it here. It will automatically appear when `contextStore.loading` is true.